libraries/protocol/steem_operations.cpp

```diff
@@ -631,6 +631,39 @@ namespace steem { namespace protocol {
       FC_ASSERT( reward_steem.amount > 0 || reward_sbd.amount > 0 || reward_vests.amount > 0, "Must claim something." );
    }
 
+#ifdef STEEM_ENABLE_SMT
+   void claim_reward_balance2_operation::add_reward_token( const asset& reward_token )
+   {
+      if( reward_token.amount <= 0 )
+      {
+         // Negative amount is invalid.
+         FC_ASSERT( reward_token.amount == 0, "Cannot claim a negative amount" );
+         // Zero amount is ineffective.
+         return;
+      }
+
+      auto insert_it = lower_bound(reward_tokens.begin(), reward_tokens.end(), reward_token, nai_less());
+      FC_ASSERT( insert_it == reward_tokens.end() || insert_it->symbol.to_nai() != reward_token.symbol.to_nai(),
+                 "Duplicate symbol ${s} inserted into claim reward operation container.", ("s", reward_token.symbol) );
+      reward_tokens.insert( insert_it, reward_token );
+   }
+
+   void claim_reward_balance2_operation::validate()const
+   {
+      validate_account_name( account );
+      FC_ASSERT( reward_tokens.empty() == false, "Must claim something." );
+      FC_ASSERT( reward_tokens.begin()->amount >= 0, "Cannot claim a negative amount" );
+      for( auto itl = reward_tokens.begin(), itr = ++itl; itr != reward_tokens.end(); ++itl, ++itr )
+      {
+         FC_ASSERT( itl->symbol.to_nai() <= itr->symbol.to_nai(), 
```

Comment:
We should make it standard practice that whenever an NAI is used in an assertion that the nai strings be returned in the error message. @youkaicountry We should create an issue to audit all SMT related error messages prior to release to enforce this standard.

Reply:
I like that idea, I'll enshrine this in an issue we can execute later.